lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -743,6 +743,41 @@ OpFoldResult MuxOp::fold(ArrayRef<Attribute> constants) {
   return {};
 }
 
+void MuxOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
+                                        MLIRContext *context) {
+  struct Folder final : public OpRewritePattern<MuxOp> {
+    using OpRewritePattern::OpRewritePattern;
+    LogicalResult matchAndRewrite(MuxOp op,
+                                  PatternRewriter &rewriter) const override {
+      auto width = op.getType().cast<IntegerType>().getWidth();
+
+      if (width == 1) {
+        APInt value;
+
+        // mux(a, true, b) -> or(a, b)
+        if (matchPattern(op.trueValue(), m_RConstant(value))) {
+          if (value == 1) {
+            SmallVector<Value, 4> newOperands{op.cond(), op.falseValue()};
```

Comment:
I'd suggest using a C array for this instead of a smallvector, e.g.:

Value newOperands[] = { op.cond(), op.falseValue()};

You can also inline that into the next line like this:

rewriter.replaceOpWithNewOp<OrOp>(op, op.getType(), { op.cond(), op.falseValue()});